### PR TITLE
Create NightVerify pipeline with Valgrind. Add CppCheck and SonarQube

### PR DESCRIPTION
**Pull request by ystefinko**
Script execute build in RelWithDebInfo mode (chosen it because there many flaky tests in Debug mode still),
then run all tests with Valgrind, generating Valgrind reports into folder reports/valgrind, save this dir as artifacts.
In next stage we perform cppcheck for code analysis, and send it reports together with Valgrind reports to SonarQube analysis.
Small refactor of names in .sh and .variables.
Small fix in echo messages in FV tests.
Several tests are filtered as they became flaky.

Relates-to: OLPEDGE-987
Signed-off-by: Yaroslav Stefinko <ystefinko@intellias.com>

### CODE REVIEW

File: scripts/linux/nv/gitlab_build_valgrind.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -e
+mkdir -p build && cd build
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -OLP_SDK_BUILD_EXAMPLES=ON -DOLP_SDK_BUILD_DOC=ON -DBUILD_SHARED_LIBS=ON ..
```
Comment (andescu) [inline scripts/linux/nv/gitlab_build_valgrind.sh]:
```suggestion
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DOLP_SDK_BUILD_EXAMPLES=ON -DOLP_SDK_BUILD_DOC=ON -DBUILD_SHARED_LIBS=ON ..
```

What about tests? Do you need to build tests for this?
If you need then better explicitly declare that with -DOLP_SDK_ENABLE_TESTING=ON.
If you do not need to build tests then disable them with -DOLP_SDK_ENABLE_TESTING=OFF

File: scripts/linux/nv/gitlab_test_valgrind.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+### Running all unit groups
+for test_name in "${TEST_TARGET_NAMES[@]}"
+do
+{
+    if [[ ${test_name} == "olp-cpp-sdk-core-test" ]] ; then
+        EXCEPTION="--gtest_filter="-DefaultCacheTest.BadPath""
+        else
+        EXCEPTION=""
+    fi
+    FLAKY_CHECK="" #FLAKY_CHECK="--gtest_repeat=10"
```
Comment (andescu) [inline scripts/linux/nv/gitlab_test_valgrind.sh]: Remove comment.

File: scripts/linux/nv/gitlab_test_valgrind.sh [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+}
+done
+
+### Running all unit groups
+for test_name in "${TEST_TARGET_NAMES[@]}"
+do
+{
+    if [[ ${test_name} == "olp-cpp-sdk-core-test" ]] ; then
+        EXCEPTION="--gtest_filter="-DefaultCacheTest.BadPath""
+        else
```
Comment (andescu) [inline scripts/linux/nv/gitlab_test_valgrind.sh]: Iteration seems to not be correct for `else`

File: scripts/linux/nv/gitlab_test_valgrind.sh [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    fi
+done
+
+echo "Summary ejected from reports is : "
+for report in `ls ${REPO_HOME}/reports/*-report.xml`
+do
+    echo -e "$(basename ${report}): \t $(cat ${report} | sed -n 2p | sed -e "s/timestamp=.*//" | sed -e "s/\<testsuites//" )"
+done
+
+echo "Artifacts download URL: https://main.gitlab.in.here.com/olp/edge/olp-sdk/olp-sdk-cpp/-/jobs/$CI_JOB_ID/artifacts/download"
```
Comment (andescu) [inline scripts/linux/nv/gitlab_test_valgrind.sh]: Remove internal URL from script!

File: scripts/linux/nv/gitlab_test_valgrind.sh (hunk 3 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_valgrind.sh]: Do you mean I need to remove 1 tab before _else_ ?

Comment (andescu) [reply scripts/linux/nv/gitlab_test_valgrind.sh]: So it looks to me.

File: scripts/linux/nv/gitlab_test_valgrind.sh (hunk 2 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_valgrind.sh]: Done.Thanks!

File: scripts/linux/nv/gitlab_test_valgrind.sh (hunk 4 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_valgrind.sh]: Done.Thanks!

File: scripts/linux/nv/gitlab_test_valgrind.sh (hunk 3 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_valgrind.sh]: Done. Thanks.

File: scripts/linux/nv/gitlab_build_valgrind.sh (hunk 1 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_build_valgrind.sh]: Done.Thanks!
Added -DOLP_SDK_ENABLE_TESTING=ON because we need them.

**Pull request merged**